### ISSUE TITLE
CSI/training/unsup/simclr_CSI.py

### ISSUE THREAD

**Issue by AtsuMiyai** [alinlab/CSI#39]
"unsup/simclr_CSI.py" is the code for unlabeled multi-class. But in this code,  you use the labeles.   

``             
outputs_aux['penultimate'] = torch.cat([penul_1, penul_2])   # only use original rotation    ``    
        ### Linear evaluation ###        
``    
outputs_linear_eval = linear(outputs_aux['penultimate'].detach())      ``        
``    
loss_linear = criterion(outputs_linear_eval, labels.repeat(2))  ``      
``    linear_optim.zero_grad()    ``    
``    loss_linear.backward()    ``    
``    linear_optim.step()    ``    
``    losses['cls'].update(0, batch_size)    ``    
``    losses['sim'].update(loss_sim.item(), batch_size)    ``    
``    losses['shift'].update(loss_shift.item(), batch_size)   ``     

Hence, I think the result with this code is wrong.   
We apologize for the inconvenience, but we would appreciate your answer.

**Comment by jihoontack** [comment]
Hi thank you for your interest in our paper,

As you can see in the following line, we `detach()` the prediction from the penultimate output.
`outputs_linear_eval = linear(outputs_aux['penultimate'].detach())
`
Hence there are no such gradient updates by the label.

Thank you,
Jihoon
